mailing_list/templatetags/enrollment_tags.py

```diff
@@ -0,0 +1,69 @@
+'''
+NOTE: These template tags were copied from icommons_lti_tools/add_people as of
+e7437aaf35f81880a19ab18f69e51f19f632ac0f, to support a stripped down version of
+add_people/templates/_enrollments_list_by_name.html.  Refactoring to split that
+template out into icommons_ui, then retesting manage sections was deemed too
+much extra work for a single reuse.  If we need to use this view of course
+enrollments again, we should do the refactor then.
+'''
+
+from django import template
+from django.template.defaultfilters import stringfilter
+
+register = template.Library()
+
+ENROLLMENT_TYPE_DICTIONARY = {
+    'TeacherEnrollment': 'Teacher',
+    'TaEnrollment': 'TA',
+    'Guest': 'Guest',
+    'ObserverEnrollment': 'Observer',
+    'DesignerEnrollment': 'Designer',
+    'StudentEnrollment': 'Student',
+    'Shopper': 'Shopper',
+    'StudentViewEnrollment': 'StudentView',
+}
+
+
+@register.filter
+def get_enroll_type_display(value):
+	return ENROLLMENT_TYPE_DICTIONARY.get(value, value)
+
+
+@register.filter(is_safe=True)
+@stringfilter
+def to_letter_range(name, args):
+    """
+    Return the letter range for a given string where the range is given as a comma separated string.  For
+    example, passing 'Doe, John' to the filter with arguments of 'A-C,D-F,G-Z' would return
+    'D-F'.  Ranges can either be in the form 'A-D' or can be single letters like 'S'.
+    """
+    try:
+        ord_val = ord(name[0].upper())
+        letter_ranges = [arg.strip().upper() for arg in args.split(',')]
+        for letter_range in letter_ranges:
+            range_len = len(letter_range)
+            if ord(letter_range[0]) <= ord_val <= ord(letter_range[range_len - 1]):
+                return letter_range
+    except IndexError:
+        # Either the name passed in was empty, or one of the ranges was passed in as an empty string, so
+        # just return an empty string in this case.
+        return ''
+
+
+@register.filter
+def list_comp(lst, arg):
+    """
+    Build a list comprehension based on the key "arg" of a given list
+    """
+    comp = [x[arg] for x in lst]
+    return comp
+
+
+@register.filter(is_safe=True)
+def enrollment_lname(enrollment):
+    try:
+        lname_first = enrollment['user']['sortable_name']
+    except KeyError:
+        return ''
+    return lname_first.split(',')[0]
+
```

Comment:
django-icommons-ui, I think...that's where we've got other shared templates, so it's where those templates' custom tags belong too.  Again, putting it there didn't seem to be worth the effort if we weren't also going to refactor manage sections to use it, which felt like scope creep.